packages/@aws-cdk/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -0,0 +1,334 @@
+  const userArnStr: string = (requestContextInterface.userArn) ? '"userArn":"$context.identity.userArn",' : '';
+  const requestIdStr: string = (requestContextInterface.requestId) ? '"requestId":"$context.requestId",' : '';
+  const resourceIdStr: string = (requestContextInterface.resourceId) ? '"resourceId":"$context.resourceId",' : '';
+  const resourcePathStr: string = (requestContextInterface.resourcePath) ? '"resourcePath":"$context.resourcePath",' : '';
+  const endStr = '}';
+
+  let requestContextString :string = contextStr + accountIdStr + apiIdStr + apiKeyStr +
+        authorizerPrincipalIdStr + callerStr + cognitoAuthenticationProviderStr + cognitoAuthenticationTypeStr +
+        cognitoIdentityIdStr + cognitoIdentityPoolIdStr + httpMethodStr + stageStr + sourceIpStr + userStr +
+        userAgentStr + userArnStr + requestIdStr + resourceIdStr + resourcePathStr + endStr;
+
+  if (requestContextString !== (contextStr + endStr)) {
+    //Removing the last comma froom the string only if it has a value inside
+    requestContextString = requestContextString.substring(0, requestContextString.length-2) + '}';
+  }
+  return requestContextString;
+}
+
+function requestContext(requestContextObj: RequestContext | undefined): string {
+  const requestContextStr: string = requestContextBuilder({
+    accountId: (requestContextObj) ? requestContextObj?.accountId : false,
+    apiId: (requestContextObj) ? requestContextObj?.apiId : false,
+    apiKey: (requestContextObj) ? requestContextObj?.apiKey : false,
+    authorizerPrincipalId: (requestContextObj) ? requestContextObj?.authorizerPrincipalId : false,
+    caller: (requestContextObj) ? requestContextObj?.caller : false,
+    cognitoAuthenticationProvider: (requestContextObj) ? requestContextObj?.cognitoAuthenticationProvider : false,
+    cognitoAuthenticationType: (requestContextObj) ? requestContextObj?.cognitoAuthenticationType : false,
+    cognitoIdentityId: (requestContextObj) ? requestContextObj?.cognitoIdentityId : false,
+    cognitoIdentityPoolId: (requestContextObj) ? requestContextObj?.cognitoIdentityPoolId : false,
+    httpMethod: (requestContextObj) ? requestContextObj?.httpMethod : false,
+    stage: (requestContextObj) ? requestContextObj?.stage : false,
+    sourceIp: (requestContextObj) ? requestContextObj?.sourceIp : false,
+    user: (requestContextObj) ? requestContextObj?.user : false,
+    userAgent: (requestContextObj) ? requestContextObj?.userAgent : false,
+    userArn: (requestContextObj) ? requestContextObj?.userArn : false,
+    requestId: (requestContextObj) ? requestContextObj?.requestId : false,
+    resourceId: (requestContextObj) ? requestContextObj?.resourceId : false,
+    resourcePath: (requestContextObj) ? requestContextObj?.resourcePath : false,
+  });
+
+  const search = '"';
+  const replaceWith = '@@';
+  return requestContextStr.split(search).join(replaceWith);
```

Comment:
Can you add some comments on what is happening here.

Reply:
done